### PR TITLE
update incomplete snapshots dark mode - RSC-1568

### PR DESCRIPTION
**Pull request by EmilyBermudez88**
https://issues.sonatype.org/browse/RSC-1568

I left out NxFilterInput (snapshots involving the clear btn), since I expect they will be dealt with in RSC-1563(https://issues.sonatype.org/browse/RSC-1563).

I have added comments into Jira tickets of components that still need to be completed, but are included in snapshots of components that are already done or have no ticket (ie. NxFilterDropdown snapshots that need to be updated in NxTable, or NxFieldset snapshots that need to be updated when NxTransferList is complete)

I did discover that `--nx-color-text-placeholder` is incorrect in dark mode. It's set to `--nx-color-indigo-70` instead of `--nx-swatch-indigo-70`, so will need to be updated, which will update snapshots. I can add that to here as part of general clean up, or I can make a separate bug ticket.

### CODE REVIEW

**Comment by rpokorny** [comment]
> I did discover that --nx-color-text-placeholder is incorrect in dark mode. It's set to --nx-color-indigo-70 instead of --nx-swatch-indigo-70, so will need to be updated, which will update snapshots. I can add that to here as part of general clean up, or I can make a separate bug ticket.

Let's do it here

**Comment by EmilyBermudez88** [comment]
>>I did discover that --nx-color-text-placeholder is incorrect in dark mode. It's set to --nx-color-indigo-70 instead of --nx-swatch-indigo-70, so will need to be updated, which will update snapshots. I can add that to here as part of general clean up, or I can make a separate bug ticket.

> Let's do it here

 45272c5

File: gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-disabled-nx-date-input-looks-disabled-when-invalid-1-dark.png
Comment (rpokorny) [inline gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-disabled-nx-date-input-looks-disabled-when-invalid-1-dark.png:1]: Pretty sure this is wrong.  See the placeholder color in the disabled text input mockups

File: gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-validatable-nx-date-input-has-invalid-validation-styles-when-invalid-1-dark.png
Comment (rpokorny) [inline gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-validatable-nx-date-input-has-invalid-validation-styles-when-invalid-1-dark.png:1]: Why aren't the validation styles showing up in this screenshot? They show up in manual testing

File: gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-validatable-nx-date-input-has-validation-styles-when-valid-1-dark.png
Comment (rpokorny) [inline gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-validatable-nx-date-input-has-validation-styles-when-valid-1-dark.png:1]: More missing validation styles

File: gallery/visualtests/__image_snapshots_dark__/nx-global-header-js-nx-global-header-with-full-content-looks-right-1-dark.png
Comment (rpokorny) [inline gallery/visualtests/__image_snapshots_dark__/nx-global-header-js-nx-global-header-with-full-content-looks-right-1-dark.png:1]: These full page global sidebar snapshots should probably just be deleted for now; the tickets for the global sidebar and global header haven't been worked yet.

Comment (EmilyBermudez88) [reply gallery/visualtests/__image_snapshots_dark__/nx-global-header-js-nx-global-header-with-full-content-looks-right-1-dark.png:1]: > These full page global sidebar snapshots should probably just be deleted for now; the tickets for the global sidebar and global header haven't been worked yet.

Makes sense! Reverted here [034c1b4](https://github.com/sonatype/sonatype-react-shared-components/pull/1065/commits/034c1b42990ea90a699b1fc339ac434bdaa88f9f)

File: gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-validatable-nx-date-input-has-invalid-validation-styles-when-invalid-1-dark.png
Comment (EmilyBermudez88) [reply gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-validatable-nx-date-input-has-invalid-validation-styles-when-invalid-1-dark.png:1]: > Why aren't the validation styles showing up in this screenshot? They show up in manual testing

Interesting, it's the same for thepre-existing light mode snapshots, so investigating further

File: gallery/visualtests/__image_snapshots_dark__/nx-global-header-js-nx-global-header-with-full-content-looks-right-1-dark.png
Comment (EmilyBermudez88) [reply gallery/visualtests/__image_snapshots_dark__/nx-global-header-js-nx-global-header-with-full-content-looks-right-1-dark.png:1]: > > Makes sense! Reverted here [034c1b4](https://github.com/sonatype/sonatype-react-shared-components/pull/1065/commits/034c1b42990ea90a699b1fc339ac434bdaa88f9f)
> 
> Not reverted, deleted. They shouldn't have been added yet in the first place.

Sorry about that, deleted [ef74cb0](https://github.com/sonatype/sonatype-react-shared-components/pull/1065/commits/ef74cb0924c1e9ac9522a1280fd3f328be0cfb8a)

File: gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-validatable-nx-date-input-has-invalid-validation-styles-when-invalid-1-dark.png
Comment (EmilyBermudez88) [reply gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-validatable-nx-date-input-has-invalid-validation-styles-when-invalid-1-dark.png:1]: > > Why aren't the validation styles showing up in this screenshot? They show up in manual testing
> 
> Interesting, it's the same for the pre-existing light mode snapshots, so investigating further

I think the `inputElement.evaluate(e => {e.value= '2019-01-01; })` wasn't working as anticipated. I tried to use that same method on the validation snapshots for NxTextInput and the validation styles also weren't applied. I looked to see if there was anything in Puppeteer that would let you select something from a date picker (like fileChooser for NxFIleUpload), but doesn't seem like there's anything. So I tried instead to mimic typing into the input instead, which did the trick. I'm not sure if this is the best approach, but not sure what else to try [cad61d2](https://github.com/sonatype/sonatype-react-shared-components/pull/1065/commits/cad61d2e668a7a877ef06474c5dcf0918a576c1e)

File: gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-disabled-nx-date-input-looks-disabled-when-invalid-1-dark.png
Comment (EmilyBermudez88) [reply gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-disabled-nx-date-input-looks-disabled-when-invalid-1-dark.png:1]: > Pretty sure this is wrong. See the placeholder color in the disabled text input mockups

[b81b10f](https://github.com/sonatype/sonatype-react-shared-components/pull/1065/commits/b81b10f7145d6373bf80af1ae048c5242d180735)So this fixes the disabled color on Chrome, Safari and Microsoft Edge, but doesn't in Firefox.

Comment (rpokorny) [reply gallery/visualtests/__image_snapshots_dark__/nx-date-input-js-nx-date-input-disabled-nx-date-input-looks-disabled-when-invalid-1-dark.png:1]: > [b81b10f](https://github.com/sonatype/sonatype-react-shared-components/pull/1065/commits/b81b10f7145d6373bf80af1ae048c5242d180735) So this fixes the disabled color on Chrome, Safari and Microsoft Edge, but doesn't in Firefox. I've been looking for a Firefox solution and all answers online seem to point towards using the -webkit-datetime-edit selector, but there is no mention of -moz-datetime-edit (or any other solution) to remedy the problem on Firefox

I think it's ok if Firefox can't do it

**Review by rpokorny — APPROVED** [review]

**Pull request merged**